lms/djangoapps/course_api/blocks/forms.py

```diff
@@ -106,20 +108,51 @@ def clean(self):
         cleaned_data['user'] = self._clean_requested_user(cleaned_data, usage_key.course_key)
         return cleaned_data
 
+    def clean_username(self):
+        if 'username' in self.data:
+            return self.cleaned_data['username']
+        else:
+            return None
+
     def _clean_requested_user(self, cleaned_data, course_key):
         """
         Validates and returns the requested_user, while checking permissions.
         """
         requesting_user = self.initial['requesting_user']
         requested_username = cleaned_data.get('username', None)
 
-        if not requested_username:
+        if requesting_user.is_anonymous:
+            return self._verify_anonymous_user(cleaned_data, course_key)
+
+        if requested_username is None:
             return self._verify_no_user(requesting_user, cleaned_data, course_key)
         elif requesting_user.username.lower() == requested_username.lower():
             return self._verify_requesting_user(requesting_user, course_key)
         else:
             return self._verify_other_user(requesting_user, requested_username, course_key)
 
+    @staticmethod
+    def _verify_anonymous_user(cleaned_data, course_key):
+        """
+        Verifies form for when the requesting user is anonymous.
+        """
+        if cleaned_data.get('all_block', None) is not None:
+            raise PermissionDenied(
+                u"Anonymous users do not have permission to access all blocks in '{course_key}'."
+                    .format(course_key=six.text_type(course_key))
+            )
+
+        if cleaned_data.get('username', None) is not '':
+            raise PermissionDenied(u"Anonymous users cannot access another user's blocks.")
+
+        if not is_course_public(get_course_by_id(course_key, depth=0)):
```

Comment:
@xitij2000, similar to my comment on the usage of `get_course_by_id()` in a review comment below, the `Http404` exception raised by `get_course_by_id()` may have to be handled if we don't want that to propagate.

Reply:
@xitij2000, just wanted to confirm if it is okay to let the `Http404` exception to propagate in this case.